### DISCUSSION TITLE
Passing data from a client component to an async server component

### DISCUSSION THREAD

**Discussion by Xexr** [vercel/next.js#43346] [Help]
I'm trying to get my head around how we should be building next 13 apps. I thought I had it, but now I've got stuck in a circle so I think my understanding must be flawed.

From reading through the docs, I thought the preferred approach was to do as much data fetching as possible inside of server components. I get how this works simplistically where you fetch initial data but I thought it extended further to then requesting further data based on user changes and refreshing data.

Using a simple example, let's say I have a multi page table of data. Getting the first page inside a server component and rendering the first page is straight forward.

But what about when the user requests the next page? Traditionally, that would then be handled as a client side request but I was under the impression that I could have a server component nested under a client component so would be able to update a prop and then the server component would rerender with the new data. This part of the docs is what led me to believe I could do this:

https://beta.nextjs.org/docs/rendering/server-and-client-components#importing-server-components-into-client-components

However, in practice I can't work out how to have an async server component nested in a client component (so I can use some of the client components state in one of the server components props).

Is this possible or have I got the wrong end of the stick and actually server components are only good for initial data fetching?

**Comment by Xexr** [comment]
Thinking on this more. Is it currently the case that the only places a server component can read from are:

1. Props passed directly from parent server components
2. Params / query param
4. Headers
5. Cookies

So going back to my multipage example above, if I wanted to reflect the changed page then I have to do it via params or headers?

Ok in this simplistic example, but say I also had a tonne of filters I wanted to apply in the server component data fetch. These would have to be encoded into params or headers, there's no other way for me to pass them? For this example, I know we don't really have best practice yet, but would people lean towards doing it this way or sticking with the data fetching being in a client component using use or react query etc.

Grateful if someone could let me know if I'm fundamentally misunderstanding something.

Many thanks.

**Reply by feliche93** [reply]
@Xexr exactly facing the same issue I’ll link to mine below, we can check each others for answers!

**Reply by feliche93** [reply]
Here’s my related question: https://github.com/vercel/next.js/discussions/43388#discussion-4601471
